Python/ceval.c

```diff
@@ -5022,12 +5022,23 @@ import_from(PyObject *v, PyObject *name)
     return x;
  error:
     pkgpath = PyModule_GetFilenameObject(v);
+    if (pkgname == NULL) {
+        pkgname_or_unknown = PyUnicode_FromString("<unknown module name>");
+    } else {
+        pkgname_or_unknown = pkgname;
+    }
```

Comment:
@Carreau Yes, I think the refcounting is right now.  It can sometimes be difficult to trace all the various exit paths to make sure that everything gets decref'd and incref'd properly, but as best I can tell, you've got it right now.  Thanks!

The only other thing is that it's *possible* for the `PyUnicode_FromString("<unknown module name>")` to fail and return NULL.  Probably the only realistic cause of that would be a memory error (not sure if an encoding error could do it given that this is an ascii C string).  If you want to be pedanditc, it would be useful to check for that error condition and do something more reasonable in that case.